src/revivalpmmp/pureentities/entity/monster/walking/Skeleton.php

```diff
@@ -110,14 +110,16 @@ public function entityBaseTick(int $tickDiff = 1) : bool{
 		// Timings::$timerEntityBaseTick->startTiming();
 
 		$hasUpdate = parent::entityBaseTick($tickDiff);
-
+		
 		$time = $this->getLevel() ? $this->getLevel()->getTime() % Level::TIME_FULL : Level::TIME_NIGHT;
 		if(
-			!$this->isOnFire()
-			&& ($time < Level::TIME_NIGHT || $time > Level::TIME_SUNRISE)
-			&& !($this->getLevel()->getBlock($this) instanceof Water)
+			!$this->isOnFire() //if not already on fire
+			&& ($time < Level::TIME_SUNSET || $time > Level::TIME_SUNRISE) // If time inferior of TIME_NIGHT and superior of TIME_SUNRISE
+			&& !($this->getLevel()->getBlock($this) instanceof Water) // IF not in water
+			&& $this->level->getBlockSkyLightAt($this->x, $this->y, $this->z) >= 14 //If is in the sun
+			&& $this->getMobEquipment()->getHelmet() == null
 		){
-			$this->setOnFire(100);
+			$this->setOnFire(2);
```

Comment:
In the future, it would be best to submit non-related changes in their own PR's.